src/sasquatchbackpack/sources.py

```diff
@@ -0,0 +1,180 @@
+from datetime import timedelta
+
+from sasquatchbackpack.scripts import usgs
+
+
+class data_source:
+    """Base class for all relevant backpack data sources
+
+    Parameters
+    ----------
+    namespace
+        Which kafka namespace should be used, see
+        https://sasquatch.lsst.io/user-guide/namespaces.html#namespaces
+    topic name
+        Specific source name, used as an identifier
+    schema directory
+        Directory path to the relevant source schema, eg:
+         "src/sasquatchbackpack/schemas/*.avsc"
+    """
+
+    def __init__(self, namespace: str, topic_name: str, schema_directory: str):
+        self.namespace = namespace
+        self.topic_name = topic_name
+        self.schema_directory = schema_directory
+        self.results: list = []
+
+    def set_new_payload_values(self) -> None:
```

Comment:
I would start simple. It seems you want the DataSource class to be responsible for querying the data and returning a formatted list of records based on a schema. 
But in `send_data()` in sasquatch.py you also use the schema to format the payload. That could be moved here as well.  You could have a `USGSDataSource` class that queries the USGS API and return the payload based on the schema. I think this is a good time to pause and rethink the design. So here's what we are going to do. I'm going to  suggest you to read https://realpython.com/solid-principles-python/ and based on that to skecth the classes you need, before writing any code. Then when you feel confortable we can get together and discuss it.